### PR TITLE
8306575: Clean up and open source four Dialog related tests

### PR DESCRIPTION
**Pull request by luchenlin**
I backport this for parity with 11.0.21-oracle.

test/jdk/java/awt/Dialog/DialogLocationTest.java
"random.nextInt(insetX, width - 300)" is modified as random.nextInt(int origin, int bound) is not supported in jdk11.

"System.out.println(message.formatted(newX, newY, eventLoc.x, eventLoc.y, eventLoc.x - newX, eventLoc.y - newY))"
is also modified as String.formatted(Object... args) is not supported in jdk11.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306575](https://bugs.openjdk.org/browse/JDK-8306575): Clean up and open source four Dialog related tests (**Bug** - P4)

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/2088/head:pull/2088` \
`$ git checkout pull/2088`

Update a local copy of the PR: \
`$ git checkout pull/2088` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/2088/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2088`

View PR using the GUI difftool: \
`$ git pr show -t 2088`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/2088.diff">https://git.openjdk.org/jdk11u-dev/pull/2088.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/2088#issuecomment-1683402946)

### CODE REVIEW

File: test/jdk/java/awt/Dialog/DialogLocationTest.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                .getLocalGraphicsEnvironment()
+                .getMaximumWindowBounds();
+        height = screen.height;
+        width = screen.width;
+        insetX = screen.x;
+        insetY = screen.y;
+
+        for (i = 0; i < 100; i++) {
+            newX = random.nextInt(width - 300) +  insetX;
+            newY = random.nextInt(height - 400) + insetY;
```
Comment (TheRealMDoerr) [inline test/jdk/java/awt/Dialog/DialogLocationTest.java:105]: Your version will get a number between insetX and insetX+width-300.
The original change: between insetX and width-300.

Comment (luchenlin) [reply test/jdk/java/awt/Dialog/DialogLocationTest.java]: fixed to
newX = random.nextInt(width - 300 - insetX) +  insetX;
newY = random.nextInt(height - 400 - insetY) + insetY;

**Review by TheRealMDoerr — APPROVED** [review]
Backport looks good. Did you test it manually?

**Comment by luchenlin** [comment]
> Backport looks good. Did you test it manually?

yes, manually test pass.

**Comment by TheRealMDoerr** [comment]
GHA Pre-submit test failures are unrelated.
/sponsor

**Pull request closed without merging**